roxie/ccd/ccdmain.cpp

```diff
@@ -1128,13 +1136,13 @@ int STARTQUERY_API start_query(int argc, const char *argv[])
                 VStringBuffer xpath("./RoxieFarmProcess[@port='%u']", port);
                 if (!topology->hasProp(xpath))
                     topology->addPropTree("./RoxieFarmProcess")->setPropInt("@port", port);
-                RoxieEndpointInfo me = {RoxieEndpointInfo::RoxieServer, 0, { (unsigned short) port, myIP }};
+                RoxieEndpointInfo me = {RoxieEndpointInfo::RoxieServer, 0, { (unsigned short) port, myIP }, 0};
                 myRoles.push_back(me);
             }
-            for (unsigned channel: slaveChannels)
+            for (std::pair<unsigned, unsigned> channel: slaveChannels)
             {
                 mySlaveEP.set(ccdMulticastPort, myIP);
-                RoxieEndpointInfo me = { RoxieEndpointInfo::RoxieSlave, channel, mySlaveEP };
+                RoxieEndpointInfo me = { RoxieEndpointInfo::RoxieSlave, channel.first, mySlaveEP, channel.second };
                 myRoles.push_back(me);
```

Comment:
minor: would this be better using emplace_back?

Reply:
I think you suggested that the first time I wrote this code, and I could not come up with any way to get it to compile.

It's code where clarity is far more important than efficiency (executed once at startup) anyway.

Reply:
` myRoles.emplace_back(RoxieEndpointInfo{ RoxieEndpointInfo::RoxieSlave, channel.first, mySlaveEP, channel.second });
`

seems to work, but as you say efficiency isn't significant in this piece of code.